pkg/maps/ctmap/ctmap.go

```diff
@@ -262,44 +282,74 @@ func doGC6(m *bpf.Map, filter *GCFilter) int {
 		if nextKeyValid != nil {
 			break
 		}
-		nextKey = tmpKey
+		// remember the last found key
+		prevKey = currentKey
+		prevKeyValid = true
+		// continue from the next key
+		currentKey = nextKey
 	}
+
+	if count == m.MapInfo.MaxEntries {
+		// TODO Add a metric we can bump and observe here.
+		log.WithError(err).Warning("Garbage collection on IPv6 CT map failed to finish")
```

Comment:
Hmm, should that be `count == m.MapInfo.MaxEntries + 1` instead, so we know for sure that there must have been a restart in between? E.g. worst case we could have deleted a full map with all invalid entries, and still trigger this warn, no?

Reply:
Scratch my comment, it's fine as is. LGTM.

Reply:
I had `count` being 0-based and while it was correct, it was misleading, fixed.

Reply:
Ok, thanks. LGTM